### PR TITLE
Pipeline for component tests in compass-manager repository

### PR DESCRIPTION
**Pull request by mvshao**
**Description**

Changes proposed in this pull request:

-  added pipeline for running component tests for compass-manager repository

### CODE REVIEW

File: prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      skip_report: false
+      decorate: true
+      path_alias: github.com/kyma-project/compass-manager
+      cluster: untrusted-workload
+      max_concurrency: 10
+      branches:
+        - ^main$
+      spec:
+        containers:
+          - image: "europe-docker.pkg.dev/kyma-project/dev/test-infra/buildpack-golang:PR-7292"
```
Comment (Ressetkk) [inline prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:23]: Please don't use PR image here.

File: prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+              - "/bin/sh"
+            args:
+              - "-c"
+              - "make test"
+            resources:
+              requests:
+                memory: 1Gi
+                cpu: 1
+            securityContext:
+              privileged: true
```
Comment (Ressetkk) [inline prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:34]: Do you need privileged context for this job? Can you provide the use-case?

File: prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        pipeline.type: "integration"
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "pre-main-compass-manager-tests"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+      optional: false
+      run_if_changed: '^(go.mod|go.sum)$|^*/(.*.go|Makefile|Dockerfile|.*.sh)'
+      skip_report: false
+      decorate: true
+      path_alias: github.com/kyma-project/compass-manager
```
Comment (Ressetkk) [inline prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:16]: I don't think you need this line. The workdir automatically is set to your source.

File: prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      branches:
+        - ^main$
+      spec:
+        containers:
+          - image: "europe-docker.pkg.dev/kyma-project/dev/test-infra/buildpack-golang:PR-7292"
+            command:
+              - "/bin/sh"
+            args:
+              - "-c"
+              - "make test"
```
Comment (VOID404) [inline prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:28]: We _could consider_ using a script in our repo here, which would make it easier to modify the pipeline if we ever need anything more advanced (like displaying coverage data)

Comment (VOID404) [inline prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:28]: Could we also display coverage data, since we generate it anyway?

Comment (mvshao) [reply prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:27]: I think is not necessary to create a separate script for testing. We can easily modify the target in the Makefile.

Comment (VOID404) [reply prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:27]: But with my coverage example - when running locally it makes sense to just use the coverage file, but in pipelines it makes more sense to display it directly after tests, with no user interaction.

Comment (mvshao) [reply prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:27]: We can also use `go test ./... -cover` statement in our Makefile, to generate coverage, but without generating the unnecessary file.

Comment (VOID404) [reply prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:27]: I personally would love a file generated in Makefile (which then gets loaded to my IDE), and some coverage display only in pipelines, but `-cover` is also fine, and no coverage is good too tbh ¯\_(ツ)_/¯

File: prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+presubmits: # runs on PRs
+  kyma-project/compass-manager:
+    - name: pre-main-compass-manager-tests
```
Comment (Ressetkk) [inline prow/jobs/kyma-project/compass-manager/compass-manager-tests.yaml:3]: name should follow pattern `pull-repo-name`.

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**